### PR TITLE
Fixes for memory leaks and app lifecycle

### PR DESCRIPTION
**Pull request by sujithkanna**
Fixes issue #[Add issue number here. If you do not solve the issue entirely, please change the message e.g. "First steps for issues #IssueNumber]

Changes: [Add here what changes were made in this issue and if possible provide links.]

Screenshots for the change:

### CODE REVIEW

**Comment by devkhan** [comment]
I don't know where the memory was leaking exactly, can you elaborate please?

**Comment by sujithkanna** [comment]
In PhimpMe activity, the activity is stored as a static Constant variable. Means the instance of the activity is stored inside itself and since it is a static variable the instance will be alive throughout the entire application lifecycle even if the activity is closed which also affects the performance of the application.

**Comment by devkhan** [comment]
Oh okay, I thought there was some problem with the back button press. A suggestion though, since these two are completely unrelated issue, you should send separate PRs for them.

**Comment by sujithkanna** [comment]
Really sorry for that, I will send them separately. Thank you!

**Comment by pa1pal** [comment]
@sujithkanna Yeah, no need to be sorry! @devkhan is right in that there should be separate PRs for separate issues as it can confuse people.

**Comment by sujithkanna** [comment]
Thank you for the clarification @pa1pal and @devkhan . I am actually new to git, learning from you guys :)

**Comment by Abhi2424shek** [comment]
It would also be great if you could update the PR body as it specifies :)

**Comment by sujithkanna** [comment]
Oopsy! I missed that one. I posted the default template comment :p will fix them and push them properly.

**Comment by Abhi2424shek** [comment]
You have the edit button on the PR body, pls use it :)

File: app/src/main/java/vn/mbm/phimp/me/PhimpMe.java [hunk 1]
```diff
@@ -798,37 +796,31 @@ public void onResume() {
...
+            alertbox.setNegativeButton(getString(R.string.cancel), new DialogInterface.OnClickListener() {
+                @Override
+                public void onClick(DialogInterface dialog, int which) {
+                    //Resume to current process
+                }
+            });
+            alertbox.create().show();
 
         }
-        return super.onKeyDown(keycode, event);
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/PhimpMe.java]: Hey why was this line removed?

Comment (sujithkanna) [inline app/src/main/java/vn/mbm/phimp/me/PhimpMe.java]: The entire method has been changed. So we are using onBackPressed instead of onKeyDown

**Comment by Abhi2424shek** [comment]
@sujithkanna base your PR on the development branch.

**Comment by Abhi2424shek** [comment]
@sujithkanna updates pls?

**Comment by Abhi2424shek** [comment]
Closing kn favour of #223
